jedi/evaluate/compiled/subprocess/__init__.py

```diff
@@ -203,18 +222,7 @@ def get_sys_path(self):
 
     def _kill(self):
         self.is_crashed = True
-        try:
-            self._process.kill()
-            self._process.wait()
-        except (AttributeError, TypeError):
-            # If the Python process is terminating, it will remove some modules
-            # earlier than others and in general it's unclear how to deal with
-            # that so we just ignore the exceptions here.
-            pass
-
-    def __del__(self):
-        if not self.is_crashed:
-            self._kill()
+        _cleanup_process(self._process, self._stderr_thread)
```

Comment:
Why not `self._cleanup_callable = weakref.finalize(self, _cleanup_process, process, t)` above and then here: `self._cleanup_callable()`?

AFAIK this can lead to calling `_cleanup_process` twice for the same thread/subprocess.

Reply:
I think you are right. This could explain why I had to catch `OSError` exceptions in `_cleanup_process`. I'll change it as you suggested.